lib/battle/ReachabilityInfo.h

```diff
@@ -30,13 +32,16 @@ struct DLL_LINKAGE ReachabilityInfo
 		bool flying = false;
 		bool ignoreKnownAccessible = false; //Ignore obstacles if it is in accessible hexes
 		bool bypassEnemyStacks = false; // in case of true will count amount of turns needed to kill enemy and thus move forward
-		std::vector<BattleHex> knownAccessible; //hexes that will be treated as accessible, even if they're occupied by stack (by default - tiles occupied by stack we do reachability for, so it doesn't block itself)
-		std::map<BattleHex, ui8> destructibleEnemyTurns; // hom many turns it is needed to kill enemy on specific hex
+		const BattleHexArray * knownAccessible; //hexes that will be treated as accessible, even if they're occupied by stack (by default - tiles occupied by stack we do reachability for, so it doesn't block itself)
+		std::array<int8_t, GameConstants::BFIELD_SIZE> destructibleEnemyTurns; // how many turns it is needed to kill enemy on specific hex (index <=> hex)
```

Comment:
You mean knownAccessible ? The only place it is initialized is in `ReachabilityInfo::Parameters` constructor via `battle::Unit::getHexes` which returns precomputed, occupied hexes by units. It might have negative values, but then `AccessibilityInfo CBattleInfoCallback::getAccessibility` does check to filter those.

Reply:
Before `destructibleEnemyTurns[-1] = X` was legal. Now it is not since arrays don't have negative indices. And arrow towers during sieges are located on such hexes.

If you are sure that this case works - then OK. Just making sure since this was a problem in the past.

Reply:
Re-checked. All places that access this container are done with pre-check for hex validity.